test/suites/integration/suite_test.go

```diff
@@ -0,0 +1,41 @@
+package integration
+
+import (
+	"testing"
+
+	"github.com/aws/karpenter/pkg/apis/provisioning/v1alpha5"
+	"github.com/aws/karpenter/pkg/cloudprovider/aws/apis/v1alpha1"
+	"github.com/aws/karpenter/pkg/test"
+	"github.com/aws/karpenter/test/pkg/environment"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var env *environment.Environment
+
+func TestAPIs(t *testing.T) {
+	RegisterFailHandler(Fail)
+	BeforeSuite(func() {
+		var err error
+		env, err = environment.NewEnvironment(t)
+		Expect(err).ToNot(HaveOccurred())
+	})
+	RunSpecs(t, "Integration Suite")
+}
+
+var _ = AfterEach(func() {
+	env.ExpectCleaned()
+})
+
+var _ = Describe("Sanity Checks", func() {
+	It("should provision nodes", func() {
+		provider := test.AWSNodeTemplate(test.AWSNodeTemplateOptions{AWS: v1alpha1.AWS{
```

Comment:
If the Karpenter being tested here doesn't have a defaultInstanceProfile, this should fail. I think it's fine to leave as is, but I think we should make sure we can test both modes eventually.